src/MatrixWrapper/Distributed.C

```diff
@@ -67,12 +67,30 @@ void RepartDistMatrix::write(const ExecutorHandler &exec_handler,
             ->convert_to(non_local.get());
     }
 
-    // overwrite column indices with global indices
     if (write_global) {
+        // overwrite non_local column indices with global indices
```

Comment:
I don't really see a simple way to unit test it for now. since we write the data to disk. So we would need to reread and test it. I did manual tests for now.